src/classes/ADDR_Account_TDTM.cls

```diff
@@ -55,14 +56,11 @@ public class ADDR_Account_TDTM extends TDTM_Runnable {
             Account acc = (Account)so;
             i++;
 
-            // BEFORE INSERT
-            if (triggerAction == TDTM_Runnable.Action.BeforeInsert) {
+            // BEFORE INSERT & BEFORE UPDATE
+            if (isBeforeContext) {
+                UTIL_Address.normalizeLineBreak(acc);
```

Comment:
I'm concerned that this logic (and in ADDR_Contact_TDTM) will run on every single update of an existing Account or Contact, even if the Mailing/Billing Street was not actually modified, causing modifications to existing addresses which could cause other triggers to execute because they detect that the address was modified. recommend splitting this logic up so that the new _normalize_ logic only runs if the record is new or the Street address was actually modified:
```suggestion
            if (TDTM_Runnable.Action.BeforeInsert) {
                UTIL_Address.normalizeLineBreak(acc);
            }
            if (TDTM_Runnable.Action.BeforeUpdate) {
                if (UTIL_Address.isAddressFieldModified('Street', acc, trigger.old[i]) {
                    UTIL_Address.normalizeLineBreak(acc);
                }
            }


// ...... in UTIL_Address
public static isAddressFieldModified(String addrFieldType, SObject newRecord, SObject oldRecord) {
    Boolean isContact = (sObj.getSObjectType() == Contact.SObjectType);
    String prefix = (isContact ? 'Mailing' : 'Billing');
    String fieldName = (String) sObj.get(prefix + addrFieldType);
    
    String newValue = (newRecord.get(fieldName) != null ? (String)newRecord.get(fieldName) : null);
    String oldValue = (oldRecord.get(fieldName) != null ? (String)oldRecord.get(fieldName) : null);
    return newValue != oldValue; 
}
```

Reply:
There is an existing method in ADDR_Account_TDTM that checks for an address change.  Is it worth checking for a change as granular as street if we've already determined there has been an address change?  I'm wondering if your concern is adequately addressed by relying on the existing method to flag a changed address?